### PR TITLE
Correctness fixes for chunk queue

### PR DESCRIPTION
**Pull request by cramforce**
- Schedules task after, not just before making body visible. Bug introduxed in #23750
- Measures total time spend in task including immediate micro tasks invoked from it.
- Does not schedule a tasks if last invocation was a while ago, but the queue unwound in between.

Related to #23618

### CODE REVIEW

**Comment by cramforce** [comment]
@zhouyx Could you take a look at the percy diff in the consent flow. Thanks!

**Comment by zhouyx** [comment]
visual diff test looks good.

File: src/chunk.js [hunk 1]
```diff
@@ -408,20 +408,43 @@ class Chunks {
...
-      dev().fine(TAG, t.getName_(), 'Chunk duration', Date.now() - before);
     } finally {
-      resolved.then(() => {
-        this.schedule_();
-      });
+      // We want to capture the time of the entire task duration including
+      // scheduled immediate (from resolved promises) micro tasks.
+      // Lacking a better way to do this we just scheduled 10 nested
+      // micro tasks.
+      resolved
```
Comment (kristoferbaxter) [inline src/chunk.js]: This is extremely hacky. I cannot come up with a better solution, but it's worth mentioning.

Comment (kristoferbaxter) [reply src/chunk.js]: Down the line, using the proposed APIs from https://github.com/WICG/main-thread-scheduling will hopefully make this code unnecessary.

**Pull request merged**